lost-and-found-service-persistence/src/test/java/cz/muni/fi/CategoryDaoImplTest.java

```diff
@@ -2,129 +2,115 @@
+        categoryDao.addCategory(null);
     }
 
-    @Test
+    @Test(expectedExceptions = IllegalArgumentException.class)
     public void updateNullCategory() {
-        assertThatThrownBy(() -> categoryDao.updateCategory(null)).hasCauseInstanceOf(IllegalArgumentException.class);
+        categoryDao.updateCategory(null);
     }
 
-    @Test
+    @Test(expectedExceptions = IllegalArgumentException.class)
     public void updateNullIdCategory() {
-        assertThatThrownBy(() -> categoryDao.updateCategory(electro)).hasCauseInstanceOf(IllegalArgumentException.class);
+        entityManager.persist(electro);
+        electro.setId(null);
+        categoryDao.updateCategory(electro);
     }
 
-    @Test
+    @Test(expectedExceptions = IllegalArgumentException.class)
     public void deleteNullCategory() {
-        assertThatThrownBy(() -> categoryDao.deleteCategory(null)).hasCauseInstanceOf(IllegalArgumentException.class);
+        categoryDao.deleteCategory(null);
     }
 
-    @Test
+    @Test(expectedExceptions = IllegalArgumentException.class)
     public void deleteNullIdCategory() {
-        assertThatThrownBy(() -> categoryDao.deleteCategory(electro)).hasCauseInstanceOf(IllegalArgumentException.class);
+        entityManager.persist(electro);
+        electro.setId(null);
+        categoryDao.deleteCategory(electro);
     }
 
-    @Test
+    @Test(expectedExceptions = IllegalArgumentException.class)
     public void getNullCategoryById() {
-        assertThatThrownBy(() -> categoryDao.getCategoryById(null)).hasCauseInstanceOf(IllegalArgumentException.class);
+        categoryDao.getCategoryById(null);
     }
 
     @Test
-    public void shouldAddCategory() {
+    public void testAddCategory() {
         categoryDao.addCategory(electro);
-        assertEquals(categoryDao.getAllCategories().size(), 1);
+        assertEquals(electro, entityManager.find(Category.class, electro.getId()));
     }
 
     @Test
-    public void shouldDeleteCategory() {
-        categoryDao.addCategory(electro);
-        categoryDao.addCategory(clothes);
-        assertEquals(categoryDao.getAllCategories().size(), 2);
+    public void testDeleteCategory() {
+        entityManager.persist(electro);
         categoryDao.deleteCategory(electro);
-        assertEquals(categoryDao.getAllCategories().size(), 1);
-
-        assertNull(categoryDao.getCategoryById(electro.getId()));
-        assertEquals(categoryDao.getCategoryById(clothes.getId()), clothes);
+        assertNull(entityManager.find(Category.class, electro.getId()));
     }
 
     @Test
-    public void shouldUpdateCategory() {
-        categoryDao.addCategory(electro);
-        String newName = "Electronics";
-
-        electro.setName(newName);
-        categoryDao.updateCategory(electro);
-
-        Category updatedCategory = categoryDao.getCategoryById(electro.getId());
+    public void testUpdateCategory() {
```

Comment:
Can you modify the test by by adding setAttribute too and make use of getters (getAttribute and getName) in verification?